core/src/main/java/io/questdb/cutlass/http/processors/IlpTudCache.java

```diff
@@ -0,0 +1,231 @@
+            WalTableUpdateDetails tud = tableUpdateDetails.get(tableName);
+            if (distressed) {
+                Misc.free(tud);
+            } else {
+                tud.rollback();
+            }
+        }
+        if (distressed) {
+            tableUpdateDetails.clear();
+        }
+    }
+
+    @Override
+    public void close() {
+        // Close happens when HTTP connection is closed
+        ObjList<Utf8String> keys = tableUpdateDetails.keys();
+        for (int i = 0, n = keys.size(); i < n; i++) {
+            Utf8Sequence tableName = tableUpdateDetails.keys().get(i);
+            WalTableUpdateDetails tud = tableUpdateDetails.get(tableName);
+            Misc.free(tud);
+        }
+        tableUpdateDetails.clear();
+        Misc.free(ddlMem);
+        Misc.free(path);
+    }
+
+    public void commitAll() throws Throwable {
+        boolean droppedTableFound;
+        do {
+            droppedTableFound = false;
+            ObjList<Utf8String> keys = tableUpdateDetails.keys();
+            for (int i = 0, n = keys.size(); i < n; i++) {
+                Utf8Sequence tableName = tableUpdateDetails.keys().get(i);
+                WalTableUpdateDetails tud = tableUpdateDetails.get(tableName);
+                try {
+                    if (!tud.isDropped()) {
+                        tud.commit(false);
+                    }
+                } catch (CommitFailedException e) {
+                    // We can ignore dropped tables and simply roll back the transaction
+                    if (!e.isTableDropped()) {
+                        // This real commit error, abort the batch commit
+                        throw e.getReason();
+                    } else {
+                        // Remove dropped table TUD from cache
+                        // and repeat the commit procedure
+                        tud.setIsDropped();
+                    }
+                }
+
+                if (tud.isDropped()) {
+                    tableUpdateDetails.remove(tableName);
+                    Misc.free(tud);
+                    droppedTableFound = true;
+                    break;
+                }
+            }
+        } while (droppedTableFound);
+    }
+
+    public WalTableUpdateDetails getTableUpdateDetails(
+            SecurityContext securityContext,
+            @NotNull LineTcpParser parser,
+            Pool<SymbolCache> symbolCachePool
+    ) throws TableCreateException {
+        int key = tableUpdateDetails.keyIndex(parser.getMeasurementName());
+        if (key < 0) {
+            return tableUpdateDetails.valueAt(key);
+        }
+
+        tableNameUtf16.clear();
+        Utf8s.utf8ToUtf16(parser.getMeasurementName(), tableNameUtf16);
+        TableToken tableToken = getOrCreateTable(securityContext, parser, tableNameUtf16);
+        if (!engine.isWalTable(tableToken)) {
+            throw parseException.of("cannot insert in non-WAL table", null);
```

Comment:
we could include the table name in the exception flyweight - this will result in a more informative error response?

Reply:
yes, table name is present in the error, this is the test case

```Java
                    assertRequestErrorContains(influxDB, points, "wal_not_here a=1,b=1 1233456\n", "{" +
                            "\"code\":\"invalid\"," +
                            "\"message\":\"failed to parse line protocol:errors encountered on line(s):\\n" +
                            "error in line 1: table: wal_not_here; cannot insert in non-WAL table\",\"line\":1,\"errorId\":");
```